src/interface/web/app/components/loginPrompt/loginPrompt.tsx

```diff
@@ -1,47 +1,536 @@
+        })
+            .then((res) => {
+                if (res.ok) {
+                    setCheckEmail(true);
+                    if (checkEmail) {
+                        setRecheckEmail(true);
+                    }
+                    return res.json();
+                } else {
+                    throw new Error("Failed to send magic link");
+                }
+            })
+            .then((data) => {
+                console.log(data);
+            })
+            .catch((err) => {
+                console.error(err);
+            });
+    }
+
+    if (props.isMobileWidth) {
+        return (
+            <Drawer open={true} onOpenChange={props.onOpenChange}>
+                <DrawerContent className={`flex flex-col gap-4 w-full mb-4`}>
+                    <div>
+                        {useEmailSignIn && (
+                            <EmailSignInContext
+                                email={email}
+                                setEmail={setEmail}
+                                checkEmail={checkEmail}
+                                setCheckEmail={setCheckEmail}
+                                setUseEmailSignIn={setUseEmailSignIn}
+                                recheckEmail={recheckEmail}
+                                setRecheckEmail={setRecheckEmail}
+                                handleMagicLinkSignIn={handleMagicLinkSignIn}
+                            />
+                        )}
+                        {!useEmailSignIn && (
+                            <MainSignInContext
+                                handleGoogleScriptLoad={handleGoogleScriptLoad}
+                                handleGoogleSignIn={handleGoogleSignIn}
+                                isLoading={isLoading}
+                                data={data}
+                                setUseEmailSignIn={setUseEmailSignIn}
+                                isMobileWidth={props.isMobileWidth}
+                            />
+                        )}
+                    </div>
+                </DrawerContent>
+            </Drawer>
+        );
+    }
+
+    return (
+        <Dialog open={true} onOpenChange={props.onOpenChange}>
+            <DialogContent
+                className={`flex flex-col gap-4 ${!useEmailSignIn ? "p-0 pb-4 m-0 max-w-xl" : "w-fit"}`}
+            >
+                <div>
+                    {useEmailSignIn && (
+                        <EmailSignInContext
+                            email={email}
+                            setEmail={setEmail}
+                            checkEmail={checkEmail}
+                            setCheckEmail={setCheckEmail}
+                            setUseEmailSignIn={setUseEmailSignIn}
+                            recheckEmail={recheckEmail}
+                            setRecheckEmail={setRecheckEmail}
+                            handleMagicLinkSignIn={handleMagicLinkSignIn}
+                        />
+                    )}
+                    {!useEmailSignIn && (
```

Comment:
Can this be simplified like so:

```suggestion
                    || (
```